packages/bsp/common/etc/default/armbian-motd.dpkg-dist

```diff
@@ -7,7 +7,7 @@
 MOTD_DISABLE=""
 ONE_WIRE=""
 HIDE_IP_PATTERN="^dummy0|^lo"
-PRIMARY_INTERFACE="$(ls -1 /sys/class/net/ | grep -E "en|eth|wl" -m 1)"
+PRIMARY_INTERFACE="$(ip route | grep '^default' | sed "s/[a-e].*dev //" | cut -d" " -f1)
```

Comment:
that sed/cut construct can be simplified to just `awk '{print $NF}'`, so:
`ip route | grep '^default' | awk '{print $NF}'`

Reply:
If you do want to keep using `sed` and `cut`, I'd remove the `[a-e]` part, as that doesn't seem to actually do anything and just confuses the issue.
(because of the preceding `grep '^default'` you already know the line starts with 'd', so matching on any letter from a-e doesn't really add anything, it's going to match from the start of the line anyway. The `sed` command can just be `sed "s/.* dev //"` to remove everything up to and including the word "dev".

Reply:
awk is difficult to use - examples:
default via 10.0.10.1 dev eth0 onlink
default via 10.0.10.1 dev br0 proto static

> matching on any letter from a-e doesn't really add anything

Correct, removing that is to go and it works. That's left over from an attempt to come up with a regex.

Reply:
> awk is difficult to use - examples:
> default via 10.0.10.1 dev eth0 onlink
> default via 10.0.10.1 dev br0 proto static

Curiosity is tearing me apart.
What will the "netstat -r" command show in this case?

Reply:
netstat also gives you interface of default route, yes, but it (can) take some time to determine its value.

Reply:
I'm hinting that this command has an output of 8 columns and can be easily analyzed using awk.
The situation when there are two default routes assumes that the Genmask, Flags columns will have different values. Or am I wrong?

Reply:
How can I make this situation?

Reply:
```
echo "$(ip route | grep '^default' | sed "s/.*dev //" | cut -d" " -f1)"
eth0
wlan0
```

Reply:
```
echo "$(ip route | awk '/default/ && (NR == 1) {print $5}')"
eth0
```

Reply:
For my situation, the main traffic goes through the interface that is printed first